### PR TITLE
Fix spelling errors, rename abstract base classes

### PR DESCRIPTION
**Pull request by f0uriest**
- Fixes a bunch of typos and spelling errors to shut up my spell checker
- Adds an underscore to abstract base class names so they get sorted first so inheritance works correctly
- Splits `Grid` class into a true ABC `_Grid` that `Grid` now inherits from.
- Fixes a bunch of misc spacing issues that were causing the alphabetizing script trouble.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -15,16 +12,6 @@
 from desc.vmec import VMECIO
 
 
-# print full tracebacks to help find sources of warnings
-def _warn_with_traceback(message, category, filename, lineno, file=None, line=None):
```
Comment (f0uriest) [inline tests/conftest.py:19]: because we raise warnings to errors, this isnt needed anymore.

**Review by ddudt — APPROVED** [review]

**Comment by dpanici** [comment]
Can we add the spell checker to the pre-commit? like https://github.com/codespell-project/codespell

**Comment by f0uriest** [comment]
> Can we add the spell checker to the pre-commit? like https://github.com/codespell-project/codespell

Probably but would prefer to do another PR for that since we'll need to come up with a list of words and patterns to ignore (think like all the vmec variable names ...)

File: desc/coils.py [hunk 2]
```diff
@@ -77,7 +76,7 @@ def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
             magnetic field at specified points, in either rpz or xyz coordinates
         """
         assert basis.lower() in ["rpz", "xyz"]
-        if isinstance(coords, Grid):
+        if hasattr(coords, "nodes"):
```
Comment (dpanici) [inline desc/coils.py:79]: is `hasattr` instead of `isinstance` better in some way, or is this just a cleaner way? Why this change is what I am asking

Comment (f0uriest) [reply desc/coils.py:79]: In this case it is "better" since it's what we actually need from the object. In general I was just trying to avoid importing things just for instance checking where possible
